nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/DeployCommand.java

```diff
@@ -537,11 +537,11 @@ public void execute(AdminCommandContext context) {
             Transaction t = deployment.prepareAppConfigChanges(deploymentContext);
             span.finish(); // next phase is launched by prepare
             Deployment.ApplicationDeployment deplResult = deployment.prepare(null, deploymentContext);
-            if(!loadOnly) {
+            if(deplResult != null && !loadOnly) {
                 // initialize makes its own phase as well
                 deployment.initialize(deplResult.appInfo, deplResult.appInfo.getSniffers(), deplResult.context);
             }
-            ApplicationInfo appInfo = deplResult.appInfo;
+            ApplicationInfo appInfo = deplResult != null ? deplResult.appInfo : null;
```

Comment:
`appInfo` is later used in `deployment.registerAppInDomainXML(appInfo, deploymentContext, t);`. From the looks of that statement I'd assume this should not be called with `null`.

Reply:
Yes, but when `appInfo` is null, action exit code is not success so the branch that uses it is not invoked (yes there are too many signalling paths in deployment process).